### ISSUE TITLE
Corrupted indentation with commentary_startofline = 1

### ISSUE THREAD

**Issue by dnezamaev** [tpope/vim-commentary#123]
TL;DR videos: [first](https://youtu.be/Cl6iN7djN7M) and [second](https://youtu.be/0Hb8Zzo1fOs)

Suppose we have C/C++ code like this, cursor on 'int x = 1' and doing motion like 'gcd' to comment two strings.

```
int main(int argc, char *argv[])
{
  int x = 1;
  //return 0;
}

```

Then we repeat same action to uncomment and **_loosing one space of indentation on each uncommenting_** on that two strings.

Important things here: 
1. b:commentary_startofline=1
2. return string is already commented without space (slashes are closely to text)
3. commentstring value does not matter, it can be reproduced with default /\*%s\*/
4. filetype does not matter, it can be reproduced on python code also
5. adding one more string after '//return' makes it even worse ([see first video](https://youtu.be/Cl6iN7djN7M))

P.S. Thank you for plugins, they are cool! =)
